src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/SecretBuildWrapper.java

```diff
@@ -72,6 +75,25 @@
         };
     }
 
+    @Override public OutputStream decorateLogger(AbstractBuild build, OutputStream logger)
+            throws IOException, InterruptedException, RunnerAbortedException {
+
+        Map<String,String> overrides = new HashMap<String, String>();
+        for (MultiBinding<?> binding : bindings) {
+            MultiBinding.MultiEnvironment environment = binding.bind(build, null, null, null);
```

Comment:
No, this is just no good. `Run.createBuildListener` calls `decorateLogger` before `BuildWrapper.setUp` is run. So we simply cannot use the `bind` method as currently designed. It needs an API change in `MultiBinding`.